php_memcached.c

```diff
@@ -356,20 +356,17 @@ char *php_memc_printable_func (zend_fcall_info *fci, zend_fcall_info_cache *fci_
 static zend_bool php_memcached_on_new_callback(zval *object, zend_fcall_info *fci, zend_fcall_info_cache *fci_cache, zend_string *persistent_id)
 {
 	zend_bool ret = 1;
-	zval retval;
-	zval params[2];
+	zval retval, id;
 
 	if (persistent_id) {
-		ZVAL_STR(&params[1], persistent_id);
+		ZVAL_STR(&id, persistent_id);
 	} else {
-		ZVAL_NULL(&params[1]);
+		ZVAL_NULL(&id);
 	}
 
-	/* Call the cb */
-	ZVAL_COPY(&params[0], object);
+	ZVAL_UNDEF(&retval);
 
-	fci->params         = params;
-	fci->param_count    = 2;
+	zend_fcall_info_argn(fci, 2, object, &id);
```

Comment:
one of the remaining failures is where params are set for another function call ... I tried to use fcall_info_args but failed, I must misunderstand something about parameters, I'm sure someone who knows could do it ...